api/v1alpha1/tiertemplate_types.go

```diff
@@ -28,7 +29,19 @@ type TierTemplateSpec struct {
 	Revision string `json:"revision"`
 
 	// Template contains an OpenShift Template to be used to provision either a user's namespace or cluster-wide resources
+	// Note: this field will be removed in favor of the new TemplateObjects below.
 	Template templatev1.Template `json:"template"`
+
+	// TemplateObjects contains list of Unstructured Objects that can be parsed at runtime and will be applied as part of the tier provisioning.
+	//
+	// NOTE: when specifying variables as part of the objects list , those concatenated as part of other strings do not need to be wrapped inside quotes,
+	// while those that are not part of other strings do need to be wrapped in single quotes. This is required otherwise the yaml parser will error while trying to parse those resources containing variables.
+	// eg: https://docs.google.com/document/d/1x5SoBT80df9fmVsaDgAE6DE7hE6lzmNIK087JUmgaJs/edit#heading=h.2iuytpfnmul5
+	//
+	// The template parameters values will be defined in the NSTemplateTier CRD.
+	// +optional
+	// +kubebuilder:pruning:PreserveUnknownFields
+	TemplateObjects []runtime.RawExtension `json:"templateObjects" protobuf:"bytes,3,rep,name=templateObjects"`
```

Comment:
```suggestion
	TemplateObjects []runtime.RawExtension `json:"templateObjects" protobuf:"bytes,3,opt,name=templateObjects"`
```
Can we use opt here instead of rep?

Reply:
`rep` can be used if we think that the objects would be repeated , i this case since its a list of Template Objects, i think rep would be fine

Reply:
I've accepted your suggestion, thanks.